src/math/casadi.hpp

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright (c) 2019 INRIA
+//
+
+#ifndef __pinocchio_math_casadi_hpp__
+#define __pinocchio_math_casadi_hpp__
+
+#include <casadi/casadi.hpp>
+
+namespace Eigen
+{
+  namespace internal
+  {
+    // Specialization of Eigen::internal::cast_impl for CppAD input types
+    template<typename Scalar>
+    struct cast_impl<casadi::SX,Scalar>
+    {
+#if EIGEN_VERSION_AT_LEAST(3,2,90)
+      EIGEN_DEVICE_FUNC
+#endif
+      static inline Scalar run(const casadi::SX & x)
+      {
+        return (Scalar)x;
```

Comment:
Eigen's implementation of cast_impl is:
```cpp
    return static_cast<NewType>(x);
```
Why do you have to C-style cast rather that C++-style cast operator ?

Reply:
I'm just calling the cast of Casadi Matrix type directly.

Reply:
I don't see any difference between the Eigen's implementation of `Eigen::internal::cast_impl<casadi::SX, Scalar>` and yours. Don't they do exactly the same ?

Reply:
Eigen's implementation of `Eigen::internal::cast_impl<casadi::SX, Scalar>` boils down to
```cpp
static inline Scalar run(const casadi::SX & x)
{
    return static_cast<Scalar>(x);
}
```